aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/options/AWSCognitoAuthSignUpOptions.kt

```diff
@@ -115,4 +116,31 @@ internal constructor(
             CognitoBuilder()
                 .apply(block).build()
     }
+
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (javaClass != other?.javaClass || other !is AWSCognitoAuthSignUpOptions) return false
+        if (!super.equals(other)) return false
+
+        if (validationData != other.validationData) return false
+        if (clientMetadata != other.clientMetadata) return false
+
```

Comment:
Did we miss `userAttributes`?

Reply:
No there's a `if (!super.equals(other)) return false` right above it